code/datums/ai/monkey/monkey_behaviours.dm

```diff
@@ -0,0 +1,262 @@
+		finish_action(controller, TRUE) //Target == owned
+
+	if(living_pawn.Adjacent(target) && isturf(target.loc) && !living_pawn.incapacitated())	// if right next to perp
+		// check if target has a weapon
+		var/obj/item/W
+		for(var/obj/item/I in target.held_items)
+			if(!(I.item_flags & ABSTRACT))
+				W = I
+				break
+
+		// if the target has a weapon, chance to disarm them
+		if(W && DT_PROB(MONKEY_ATTACK_DISARM_PROB, delta_time))
+			living_pawn.a_intent = INTENT_DISARM
+			monkey_attack(controller, target, delta_time)
+		else
+			living_pawn.a_intent = INTENT_HARM
+			monkey_attack(controller, target, delta_time)
+
+
+/datum/ai_behavior/monkey_attack_mob/finish_action(datum/ai_controller/controller, succeeded)
+	. = ..()
+	controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET] = null
+
+/// attack using a held weapon otherwise bite the enemy, then if we are angry there is a chance we might calm down a little
+/datum/ai_behavior/monkey_attack_mob/proc/monkey_attack(datum/ai_controller/controller, mob/living/target, delta_time)
+
+	var/mob/living/living_pawn = controller.pawn
+
+	if(living_pawn.next_move > world.time)
+		return
+
+	living_pawn.changeNext_move(CLICK_CD_MELEE) //We play fair
+
+	var/obj/item/weapon = locate(/obj/item) in living_pawn.held_items
+
+	living_pawn.face_atom(target)
+
+	// attack with weapon if we have one
+	if(weapon)
+		weapon.melee_attack_chain(living_pawn, target)
+	else
+		target.attack_paw(living_pawn)
+
+	// no de-aggro
+	if(controller.blackboard[BB_MONKEY_AGRESSIVE])
+		return
+
+	if(DT_PROB(MONKEY_HATRED_REDUCTION_PROB, delta_time))
+		controller.blackboard[BB_MONKEY_ENEMIES][target]--
+
+	// if we are not angry at our target, go back to idle
+	if(controller.blackboard[BB_MONKEY_ENEMIES][target] <= 0)
+		var/list/enemies = controller.blackboard[BB_MONKEY_ENEMIES]
+		enemies.Remove(target)
+		if(controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET] == target)
+			finish_action(controller, TRUE)
+
+/datum/ai_behavior/disposal_mob
+	behavior_flags = AI_BEHAVIOR_REQUIRE_MOVEMENT | AI_BEHAVIOR_MOVE_AND_PERFORM //performs to increase frustration
+
+/datum/ai_behavior/disposal_mob/finish_action(datum/ai_controller/controller, succeeded)
+	. = ..()
+	controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET] = null //Reset attack target
+	controller.blackboard[BB_MONKEY_DISPOSING] = FALSE //No longer disposing
+	controller.blackboard[BB_MONKEY_TARGET_DISPOSAL] = null //No target disposal
+
+/datum/ai_behavior/disposal_mob/perform(delta_time, datum/ai_controller/controller)
+	. = ..()
+
+	if(controller.blackboard[BB_MONKEY_DISPOSING]) //We are disposing, don't do ANYTHING!!!!
+		return
+
+	var/mob/living/target = controller.blackboard[BB_MONKEY_CURRENT_ATTACK_TARGET]
+	var/mob/living/living_pawn = controller.pawn
+
+	controller.current_movement_target = target
+
+	if(target.pulledby != living_pawn && !HAS_AI_CONTROLLER_TYPE(target.pulledby, /datum/ai_controller/monkey)) //Dont steal from my fellow monkeys.
```

Comment:
Why not check target is literally a monkey?

Reply:
I believe it's done so player-controller monkey can't save you from being disposed.

Reply:
It makes more sense for monkeys to not attack fellow monkeys not monkey controllers.

Reply:
Makes sense, but balance?

Reply:
eek ook eeee
monky does not care about balance
it's probably so one controller does not attack second controller